### PR TITLE
Unittests for meta- and resultnode

### PR DESCRIPTION
**Pull request by rayklauck**
Adding unittests for MetaNode and ResultNode.

### CODE REVIEW

File: tests/nodetree/test_node_unit.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+        path = join_path(next_segment)
+
+        with patch('labone.nodetree.node.ResultNode.is_child_node') as child_patch:
+            with patch('labone.nodetree.node.split_path',
+                       return_value=next_segment) as split_patch:
+                MockResultNode(()).__contains__(AnnotatedValue(path=path, value=0))
+                split_patch.assert_called_once_with(path)
+                child_patch.assert_called_once_with(next_segment)
+
+    def test_call(self):
```
Comment (tobiasah) [inline tests/nodetree/test_node_unit.py:204]:
```suggestion
    def test_call_raises(self):
```

File: tests/nodetree/test_node_unit.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        with pytest.raises(LabOneInappropriateNodeTypeError):
+            MockResultNode(()).__call__()
+
+    def test_str(self, result_node):
+        assert isinstance(str(result_node), str)
+
+    def test_repr(self, result_node):
+        assert isinstance(repr(result_node), str)
```
Comment (tobiasah) [inline tests/nodetree/test_node_unit.py:212]: issue: These two test do not make sense ... python will ensure by its own that a string is returned. Better check if the returned string contains the LabOne node path or other things

File: tests/nodetree/test_node_unit.py [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+                              ('/a/b/c', {3}),
+                              ('/a/b/e', set())
+                              ])
+    def test_results(self, path, expected):
+        value_structure = {'/a': 1, '/a/b': 2, '/a/b/c': 3, '/a/b/d': 4, '/a/b/d/e':5}
+        node = MockResultNode(tuple(split_path(path)))
+        node._value_structure = value_structure
+
+        assert set(node.results()) == expected
+
```
Comment (tobiasah) [inline tests/nodetree/test_node_unit.py:229]: issue: there is a results test missing for a subnode.  e.g. node.a.b.results() should only contain return the results behind b

File: tests/nodetree/test_node_unit.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+                             [('/a', {1,2,3,4,5}),
+                              ('/b', set()),
+                              ('/a/b', {2,3,4,5}),
+                              ('/a/b/d', {4,5}),
+                              ('/a/b/d/e', {5}),
+                              ('/a/b/c', {3}),
+                              ('/a/b/e', set())
+                              ])
+    def test_results(self, path, expected):
+        value_structure = {'/a': 1, '/a/b': 2, '/a/b/c': 3, '/a/b/d': 4, '/a/b/d/e':5}
```
Comment (tobiasah) [inline tests/nodetree/test_node_unit.py:224]: suggestion: I would change this to something that would actually work 😇 

```suggestion
        value_structure = {'/a/b': 1, '/a/c/d': 2, '/a/c/e': 3, '/a/c/f/g':4, '/a/c/f/h':5}
```

File: tests/nodetree/conftest.py [hunk 5]
```diff
@@ -28,29 +44,37 @@ def structure(self):
...
+    return {ann.path: ann for ann in zi_get_responses}
 
-    def wrapper(*args, **kwargs):
-        key = (args, frozenset(kwargs.items()))
-        if key not in cache_dict:
-            cache_dict[key] = func(*args, **kwargs)
-        return cache_dict[key]
+@pytest.fixture()
+def device_id():
+    return "dev12084"
```
[the remark this answers is not in the document]
Comment (rayklauck) [reply tests/nodetree/conftest.py:63]: actually, the tests operate on a preloaded node-structure from exactly this device. Therefore, the paths start with this specific id

[the remark this answers is not in the document]
Comment (tobiasah) [reply tests/nodetree/conftest.py:63]: but you can change the json to also use `dev1234` similar to what we use in toolkit

**Review by tobiasah — APPROVED** [review]
🥇

**Pull request merged**